### PR TITLE
Refactor url service and sessions_controller to support those changes.

### PR DESCRIPTION
**Pull request by saneshark**
## Description of change
Cleanup of URL Service and Sessions Controller logic.

## Testing done
Local Testing and Specs only.

## Testing planned
None, maybe some verification on staging that everything works as expected.

#### Unique to this PR

- [ ] _Replace this line with individual tasks unique to your PR_

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: app/controllers/v0/sessions_controller.rb [hunk 1]
```diff
@@ -19,37 +15,19 @@ class SessionsController < ApplicationController
...
-        format.html { redirect_to url }
-        format.json { render json: { url: url } }
+      type = params[:signup] ? 'signup' : params[:type]
+      if REDIRECT_URLS.include?(type)
+        url = url_service.send("#{type}_url")
+        if type == 'slo'
+          Rails.logger.info('SSO: LOGOUT', sso_logging_info)
+          reset_session
+        end
+        redirect_to url
```
Comment (saneshark) [inline app/controllers/v0/sessions_controller.rb]: @narin i would just tack the clientID on the end of URL here, no point in going through URLService.

File: spec/controllers/v0/sessions_controller_spec.rb [hunk 2]
```diff
@@ -67,29 +68,24 @@ def verify_session_cookie
...
-              expect(response).to have_http_status(:ok)
-              expect(JSON.parse(response.body).keys).to eq %w[url]
-            end
-
             it 'redirects' do
               get(:new, type: type)
               expect(response).to have_http_status(:found)
+              expect(response.location)
+                .to be_an_idme_saml_url('https://api.idmelabs.com/saml/SingleSignOnService?SAMLRequest=')
+                .with_relay_state('originating_request_id' => nil)
```
Comment (saneshark) [inline spec/controllers/v0/sessions_controller_spec.rb]: @narin for these specs, you can use just add a .with_param and test out the clientID is getting set but you'll have to setup the request to receive that param as well, so you change get(:new, type: type) to be get(:new, type: type, ga_id: '12312313') and .with_param('ga_id' => '12312313') for all of the new actions. You'll not that there are two sections where these are tested (once for when you are signed in and once for when you are not).

File: app/controllers/v0/sessions_controller.rb (hunk 1 above)
Comment (narin) [reply app/controllers/v0/sessions_controller.rb]: 👍🏼 I'll wait for this PR to get merged and then tack on the clientID.

File: lib/saml/url_service.rb [hunk 3]
```diff
@@ -84,8 +90,8 @@ def slo_url
       logout_request = OneLogin::RubySaml::Logoutrequest.new
       # cache the request for session.token lookup when we receive the response
       SingleLogoutRequest.create(uuid: logout_request.uuid, token: session.token)
-      Rails.logger.info "New SP SLO for userid '#{session.uuid}'"
```
Comment (saneshark) [inline lib/saml/url_service.rb]: accidentally removed this, will add back (at least for now).

File: app/controllers/v0/sessions_controller.rb [hunk 4]
```diff
@@ -19,37 +15,19 @@ class SessionsController < ApplicationController
...
-              Rails.logger.info('SSO: LOGOUT', sso_logging_info)
-              reset_session
-              logout_url
-            end
-      respond_to do |format|
-        format.html { redirect_to url }
-        format.json { render json: { url: url } }
+      type = params[:signup] ? 'signup' : params[:type]
+      if REDIRECT_URLS.include?(type)
+        url = url_service.send("#{type}_url")
```
Comment (U-DON) [inline app/controllers/v0/sessions_controller.rb]: :+1:

File: spec/lib/saml/url_service_spec.rb [hunk 5]
```diff
@@ -10,59 +11,143 @@
...
           expect(user.authn_context).to eq('http://idmanagement.gov/ns/assurance/loa/1/vets')
           expect_any_instance_of(OneLogin::RubySaml::Settings)
             .to receive(:authn_context=).with('http://idmanagement.gov/ns/assurance/loa/3/vets')
-          expect(subject.idme_loa3_url).to include('https://api.idmelabs.com/saml/SingleSignOnService?SAMLRequest=')
+          expect(subject.verify_url)
+            .to be_an_idme_saml_url('https://api.idmelabs.com/saml/SingleSignOnService?SAMLRequest=')
+            .with_relay_state('originating_request_id' => '123')
+        end
+
+        it 'has sign in url: with (default authn_context) first time verification' do
```
Comment (U-DON) [inline spec/lib/saml/url_service_spec.rb]: What's the difference between this test and the the one above?

File: spec/lib/saml/url_service_spec.rb [hunk 6]
```diff
@@ -10,59 +11,143 @@
...
           allow(user).to receive(:authn_context).and_return('myhealthevet')
           expect_any_instance_of(OneLogin::RubySaml::Settings)
             .to receive(:authn_context=).with('myhealthevet_loa3')
-          expect(subject.idme_loa3_url).to include('https://api.idmelabs.com/saml/SingleSignOnService?SAMLRequest=')
+          expect(subject.verify_url)
+            .to be_an_idme_saml_url('https://api.idmelabs.com/saml/SingleSignOnService?SAMLRequest=')
+            .with_relay_state('originating_request_id' => '123')
+        end
+
+        it 'has sign in url: with (myhealthevet authn_context) first time verification' do
```
Comment (U-DON) [inline spec/lib/saml/url_service_spec.rb]: Similarly, this looks identical to the test above.

**Pull request merged**